### PR TITLE
rewrite wifi.save and restore to use the storage lib (imp #1380)

### PR DESCRIPTION
**Pull request by MaBecker**
* use .wificfg to store wifi credentials
* free wifi save page and add to save_code
* remove flash page 0x77000 and add to flash_available
* move wifi restore after JavaScript environment is up and running

### CODE REVIEW

**Comment by MaBecker** [comment]
Possible to merge?

**Comment by gfwilliams** [comment]
Looks good - thanks :)

File: boards/ESP8266_BOARD.py [hunk 1]
```diff
@@ -46,8 +46,8 @@
   'saved_code' : {
     'address' : 0x78000,
     'page_size' : 4096,
-    'pages' : 3,
-    'flash_available' : 468, # firmware can be up to this size
+    'pages' : 4,
+    'flash_available' : 472, # firmware can be up to this size
```
Comment (wilberforce) [inline boards/ESP8266_BOARD.py]: I understand the code size has increased by a page, however don't understand why the firmware can be larger?

Comment (MaBecker) [reply boards/ESP8266_BOARD.py]: Gave up one Wifi page and one flash, for details see #1380

Comment (wilberforce) [reply boards/ESP8266_BOARD.py]: > espruino_esp8266_user1.bin, max 118 pages  = max size 472 KB

Where did the one for flash come from? Isn't the phy data stored in the last block by the operating system?

**Pull request merged**